File: client/src/utils/validators.js
export const composeValidators =
  (...validators) =>
  (value) =>
    validators.reduce(
      (error, validator) => error || validator(value),
      undefined,
    );

export function usernameRegexValidator(value) {
  const regex = /^[a-zA-Z][a-zA-Z0-9]{4,12}$/;

  if (!regex.test(value)) {
    return 'El usuario debe tener entre 5 y 12 caracteres. Debe iniciar con una letra. Solo letras y números permitidos';
  }
}

export function dpiRegexValidator(value) {
  const regex = /^[1-9][0-9]{12}$/;

  if (!regex.test(value)) {
    return 'El DPI debe contener 13 dígitos';
  }
}

export function nitRegexValidator(value) {
  const regex = /^[1-9][0-9]{8}$/;

  if (!regex.test(value)) {
    return 'El NIT solo debe contener 9 dígitos';
  }
}

export function phoneRegexValidator(value) {
  const regex = /^\+?1?\d{8,15}$/;

  if (!regex.test(value)) {
    return 'Entre 8 y 15 dígitos';
  }
}

File: api/utils/validators.py
""" Data validators """

# Django
from django.core.validators import RegexValidator


def dpi_regex_validator():
    dpi_regex = RegexValidator(
        regex=r'[1-9][0-9]{12}$',
        message="El DPI debe tener 13 dígitos"
    )
    return dpi_regex


def nit_regex_validator():
    nit_regex = RegexValidator(
        regex=r'[1-9][0-9]{8}$',
        message="El Nit debe tener 9"
    )
    return nit_regex


def username_regex_validator():
    username_regex = RegexValidator(
        regex=r'[a-zA-Z][a-zA-Z0-9]{4,12}$',
        message="Entre 5 y 12 caracteres iniciando con una letra"
    )
    return username_regex


def phone_regex_validator():
    phone_regex = RegexValidator(
        regex=r'\+?1?\d{8,15}$',
        message="Mínimo 8 dígitos para el número de teléfono"
    )
    return phone_regex
